### PR TITLE
Firebase

### PR DESCRIPTION
**Pull request by LucaCarroz**
Firebase bootcamp of week#02. 
Adds a new activity accessible from the main activity that allow to "get" and "set" <key, value> pairs stored in a real time database, provided by Firebase.

### CODE REVIEW

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Test
+    fun phoneNumberWrittenCorrectly() {
+        onView(withId(R.id.phoneDB))
+            .perform(typeText("1112223344"))
+            .check(matches(withText("1112223344")))
+    }
+
+    @Test
+    fun getWithInexistantKeyReturnsNothing() {
```
Comment (2Tricky4u) [inline app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt:58]: I think the name of phoneDB could be rename as it is an id of a text box

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Before
+    fun before() {
+        // Change the used database to the local database
+        Database.db = LocalDataBase()
+    }
+
+    @Test
+    fun emailWrittenCorrectly() {
+        onView(withId(R.id.emailDB))
```
Comment (2Tricky4u) [inline app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt:45]: I think the name of emailDB could be rename as it is an id of a text box

File: app/src/main/res/layout/activity_database.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (2Tricky4u) [inline app/src/main/res/layout/activity_database.xml:1]: On all this file Cirrus CI warns us with some messages like: "Missing `autofillHints` attribute" or "Hardcoded string "Get", should use `@string` resource". I think those warnings are founded and could be fixed =D

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -21,13 +21,24 @@
 
     <Button
         android:id="@+id/mainButton"
-        android:layout_width="211dp"
-        android:layout_height="73dp"
-        android:layout_marginTop="240dp"
+        android:layout_width="104dp"
```
Comment (2Tricky4u) [inline app/src/main/res/layout/activity_main.xml:24]: Here in this files we also have a lot of Cirrus warnings

**Review by 2Tricky4u — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class DatabaseActivityTest {
+    @get:Rule
+    val activityRule = ActivityScenarioRule(DatabaseActivity::class.java)
+
+    /*
+    Implementation of a local database
+     */
+    class LocalDataBase : Database() {
```
[the remark this answers is not in the document]
Comment (LucaCarroz) [reply app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt:26]: Makes sense, thanks!

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt (hunk 1 above)
Comment (LucaCarroz) [reply app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt:58]: I renamed all of the Buttons and EditTexts to follow the convention set in the main activity, thanks for pointing out that inconsistency!

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt (hunk 2 above)
Comment (LucaCarroz) [reply app/src/androidTest/java/com/github/sdp_begreen/begreen/DatabaseActivityTest.kt:45]: I renamed all of the Buttons and EditTexts to follow the convention set in the main activity, thanks for pointing out that inconsistency!

**Review by 2Tricky4u — APPROVED** [review]
👍

**Pull request merged**